src/components/ArticleDetail/ArticleComments/PostComment/PostComment.jsx

```diff
@@ -0,0 +1,51 @@
+import './PostComment.css';
+import React, { useState } from 'react';
+import { postNewComment } from '../../../../utils/api'; 
+
+const PostComment = ({ articleId, addComment }) => {
+    const [username, setUsername] = useState('');
+    const [comment, setComment] = useState('');
+    const [errorMessage, setErrorMessage] = useState('');
+
+    const handlePostComment = async (event) => {
+        event.preventDefault();
+
+        if (!username.trim() || !comment.trim()) {
+            setErrorMessage('Please fill in both fields.');
+            return;
+        }
+        
+        try {
+            const newComment = await postNewComment(articleId, username, comment);
+            addComment(newComment); 
```

Comment:
Nothing wrong with your method here but unless we need to use this function somewhere else, we don't actually need to declare this function in the parent and pass it down - we can just pass the setstate down and directly add to the comments here :)